### ISSUE TITLE
Search results always empty after apparently  successful install

### ISSUE THREAD

**Issue by leforestier** [twain47/Nominatim#339]
Hi,

I followed the installation instructions here: http://wiki.openstreetmap.org/wiki/Nominatim/Installation

I managed to run a few installs that did not generate errors, using this Luxembourg extract http://download.geofabrik.de/europe/luxembourg-latest.osm.bz2, then I tried with this pbf file: http://download.geofabrik.de/europe/liechtenstein-latest.osm.pbf . I always get the same problem: no search results found even for the simplest queries.

For example, after importing the Luxembourg extract, installing everything and running the site on localhost. 

http://127.0.0.1/nominatim/search.php?q=luxembourg

returns a HTML page that says
"No search results found"

Same with the Liechtenstein extract and the query:

http://127.0.0.1/nominatim/search.php?q=vaduz

I didn't get any error message at any of the installation steps.

After the installation with the Liechtenstein file, the `place` and `placex` tables look like they have data, but the `search_name` table is empty.

    nominatim=# SELECT count(*) FROM place;
     count 
    -------
      4774
    (1 row)

    nominatim=# SELECT count(*) FROM placex;
     count 
    -------
     37148
    (1 row)

    nominatim=# SELECT count(*) FROM search_name;
     count 
    -------
         0
    (1 row)

My configuration is postgresql 9.3, postgis 2.1 (compiled with raster support), on FreeBSD 10.2-RELEASE-p7. I'm running the nominatim website with nginx + php-fpm.
It's really difficult for me to find the cause of the problem since none of the installation steps generated an error. Could you help me find this out?

**Comment by lonvia** [comment]
Can you provide the complete install log?

**Comment by leforestier** [comment]
Here's the output of the command:

    /usr/bin/env php -f utils/setup.php -- -v --osm-file ~/planet-data/liechtenstein-latest.osm.pbf --all --osm2pgsql-cache 12261

Complete log of this command: http://www.yattag.org/static/nominatim-install/6-setup-osm-file/nominatim-server.80dd1018-4c42-4964-8aff-ea499019b652.log

(I had to use `/usr/bin/env php -f` to run the script because the shebang `#!/usr/bin/php -Cq` at the beginning of setup.php wasn't working. On FreeBSD, the php interpreter is in /usr/local/bin/, not /usr/bin)

Here's the output of running:

    cd ~/src/Nominatim &&
    /usr/bin/env php -f utils/specialphrases.php -- --countries > data/specialphrases_countries.sql &&
    psql -d nominatim -f data/specialphrases_countries.sql

The log: http://www.yattag.org/static/nominatim-install/7-add-special-phrases-countries/nominatim-server.80dd1018-4c42-4964-8aff-ea499019b652.log

(note, I'm doing everything with a single user who has superuser rights on the database and is also the user who runs the php site. Not very secure but I ended up doing this to make sure the problem wasn't just a permission issue.)

    cd ~/src/Nominatim &&
    /usr/bin/env php -f utils/specialphrases.php -- --wiki-import > data/specialphrases.sql &&
    psql -d nominatim -f data/specialphrases.sql

Log: http://www.yattag.org/static/nominatim-install/8-add-special-phrases-wiki-import/nominatim-server.80dd1018-4c42-4964-8aff-ea499019b652.log

I put the logs of the complete install here: http://www.yattag.org/static/nominatim-install/

**Comment by leforestier** [comment]
Postgresql also sent some output to syslog during the install. I found these in the log:

    Dec  6 00:39:52 j16 postgres[39183]: [6-1] ERROR:  role "mqmgr" does not exist
    Dec  6 00:39:52 j16 postgres[39183]: [6-2] STATEMENT:  ALTER TABLE public.us_postcode OWNER TO mqmgr;
    Dec  6 00:39:55 j16 postgres[39203]: [6-1] ERROR:  sequence "seq_postcodes" does not exist
    Dec  6 00:39:55 j16 postgres[39203]: [6-2] STATEMENT:  DROP SEQUENCE seq_postcodes;
    Dec  6 00:39:55 j16 postgres[39203]: [7-1] ERROR:  table "import_polygon_error" does not exist
    Dec  6 00:39:55 j16 postgres[39203]: [7-2] STATEMENT:  drop table import_polygon_error;
    Dec  6 00:39:55 j16 postgres[39203]: [8-1] ERROR:  table "import_polygon_delete" does not exist
    Dec  6 00:39:55 j16 postgres[39203]: [8-2] STATEMENT:  drop table import_polygon_delete;
    Dec  6 00:39:56 j16 postgres[39203]: [9-1] ERROR:  sequence "file" does not exist
    Dec  6 00:39:56 j16 postgres[39203]: [9-2] STATEMENT:  drop sequence file;

Not sure if these are normal or not.
Thanks for your help.

**Comment by lonvia** [comment]
The setup command didn't do anything interesting, although there is no indication in the log file as to why. It looks like it just stopped halfway through the process. Try remvoing line `ALTER TABLE public.us_postcode OWNER TO mqmgr;` from `data/us_postcode.sql` and see if you get any further. 

The setup is only successfully completed when you see as final lien:

**Comment by leforestier** [comment]
The logs seem to be very unreliable. It seems a lot of the `echo` messages in setup.php are being overwritten by other messages from the psql or osm2pgsql subprocesses.
I wrote a separate log function to log some messages to a separate file and I actually reach the "Setup finished" message.
Commenting the line `ALTER TABLE public.us_postcode OWNER TO mqmgr;` doesn't make any difference (by the way, where does this mqmgr user come from? this is the only reference to this user in the entire project)

What tables should be filled by the setup.php script? Is it normal that the `search_name` table is empty after running:
    
    /usr/bin/env php -Cq -f utils/setup.php -- -v --osm-file ~/planet-data/liechtenstein-latest.osm.pbf --all --osm2pgsql-cache 12261

(I also tried to lower the `--osm2pgsql-cache` value in case it was a lack of memory issue but the result is the same)

Thanks for your help.
